ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -85,18 +85,30 @@
   private final ReentrantLock lock = new ReentrantLock();
 
   /**
-   * Constructor. It fetches the full account metadata set and caches locally.
-   * This call is blocking until it fetches all the {@link Account} metadata from {@link HelixPropertyStore}.
+   * <p>
+   *   Constructor. It fetches the remote account data in {@code ZooKeeper} and caches locally during initialization,
+   *   and updates the remote account data during updating accounts. If a non-null {@link Notifier} is provided, it
+   *   will listen to the changes of the remote accounts, and reactively updates its local cache. It als sends message
+   *   to notify other listeners after each update made to the remote accounts.
+   * </p>
+   * <p>
+   *   This call is blocking until it fetches all the {@link Account} metadata from {@link HelixPropertyStore}.
+   * </p>
    * @param helixStore A {@link HelixPropertyStore} used by the {@code HelixAccountService}. Cannot be {@code null}.
    * @param accountServiceMetrics {@link AccountServiceMetrics} to report metrics. Cannot be {@code null}.
    * @param notifier A {@link Notifier} that will be used to publish message after updating {@link Account}s, and
-   *                 listen to {@link Account} change messages. Cannot be {@code null}.
+   *                 listen to {@link Account} change messages. Can be {@code null}.
    */
   HelixAccountService(HelixPropertyStore<ZNRecord> helixStore, AccountServiceMetrics accountServiceMetrics,
       Notifier<String> notifier) {
     this.helixStore = Objects.requireNonNull(helixStore, "helixStore cannot be null");
     this.accountServiceMetrics = Objects.requireNonNull(accountServiceMetrics, "accountServiceMetrics cannot be null");
-    this.notifier = Objects.requireNonNull(notifier, "notifier cannot be null");
+    this.notifier = notifier;
+    if (notifier == null) {
+      logger.warn("Notifier is null. Account updates cannot be notified to other entities. Local account cache may not "
```

Comment:
This is because without a `notifier` the `HelixAccountService` will not perform the full functionalities we expect. However it is still able to interact directly with the storage of account data.